src/components/SourceTabs.js

```diff
@@ -249,7 +250,10 @@ const SourceTabs = React.createClass({
           pretty: isPretty({ url: source.get("url") })
         }),
         key: source.get("id"),
-        onClick: () => selectSource(source.get("id")),
+        onClick: () => {
+          selectSource(source.get("id"));
+          this.props.highlightSource(source.get("url"));
```

Comment:
I think this feature is very similar to `Reveal` in that we need to use redux to pass the url to `SourcesTree.js`. Then, we can get a list of sources' items in `SourcesTree.js` and pass them to `ManagedTree.js` because that is where we know which items/sources are expanded. 

In `ManagedTree.js`, we're not expanding directories. We're actually checking which directories are opened, so we know which directory or file we need to highlight. It behaves the same as Atom.